plugins/Live/Live.php

```diff
@@ -114,4 +122,91 @@ public function renderVisitorIcons(&$renderedDetails, $visitorDetails)
             $renderedDetails .= $instance->renderIcons($visitorDetails);
         }
     }
+
+    public static function getSegmentWithVisitorId()
+    {
+        static $cached = null;
+        if ($cached === null) {
+            $segment = Request::getRawSegmentFromRequest();
+            if (!empty($segment)) {
+                $segment = urldecode($segment) . ';';
+            }
+
+            $idVisitor = Common::getRequestVar('visitorId', false);
+            if ($idVisitor === false) {
+                $idVisitor = Request::processRequest('Live.getMostRecentVisitorId');
+            }
+
+            $cached = urlencode($segment . 'visitorId==' . $idVisitor);
+        }
+        return $cached;
+    }
+
+
+    /**
+     * Returns all available profile summaries
+     *
+     * @return ProfileSummaryAbstract[]
+     * @throws \Exception
+     */
+    public static function getAllProfileSummaryInstances()
```

Comment:
Ideally this would be moved into a class named eg `LiveProvider` or `ProfileProvider` + not be static to be consistent with core.  Also applies to `getAllProfileSummaryClasses `.

The plugin class itself should have only a few lines of code related to events/hooks...